File: backend/GameApp.Application/Services/EnemyServices/EnemyGetService.cs
using GameApp.Application.UseCases.EnemyUseCases;
using GameApp.Domain.Entities;
using GameApp.Domain.Repositories;
using GameApp.Domain.ValueObjects.Enemies;

namespace GameApp.Application.Services.EnemyServices;


public class EnemyGetService : EnemyGetUseCase
{
    private readonly IEnemyRepository _repo;

    public EnemyGetService(IEnemyRepository repo) => _repo = repo;

    public async Task<Enemy?> GetEnemy(Guid id)
    {
        return await _repo.FetchByIdAsync(id);
    }

    public async Task<Enemy?> GetEnemyByName(EnemyName name)
    {
        return await _repo.FetchByName(name);
    }

    public async Task<IEnumerable<Enemy>> GetAllEnemys()
    {
        return await _repo.FetchAllAsync();
    }
}

File: frontend/gameapp-frontend/src/application/services/enemy-services/enemy-get-service.ts
import { Enemy } from "../../../domain/entities/enemy";
import type { IEnemyGetUseCase } from "../../usecases/enemy-use-cases/enemy-get-use-case";
import type { IEnemyRepository } from "../../repositories/ienemy-repository";
import type { EnemyName } from "../../../domain/value-objects/enemies/enemy-name";

export class EnemyGetService implements IEnemyGetUseCase {
    constructor(private enemyRepository: IEnemyRepository) {}

    async getEnemy(id: string): Promise<Enemy> {
        return await this.enemyRepository.fetchById(id);
    }

    async getEnemyByName(enemyName: EnemyName): Promise<Enemy> {
        return await this.enemyRepository.fetchByName(enemyName);
    }

    async getAllEnemys(): Promise<Enemy[]> {
        return await this.enemyRepository.fetchAll();
    }
}
